client/src/store/utils/reducerFunctions.js

```diff
@@ -13,9 +13,11 @@ export const addMessageToStore = (state, payload) => {
 
   return state.map((convo) => {
     if (convo.id === message.conversationId) {
-      convo.messages.push(message);
-      convo.latestMessageText = message.text;
-      return convo;
+      const convos = { ...convo };
+      convos.messages.push(message);
+      convos.latestMessageText = message.text;
+
+      return convos;
```

Comment:
A better name will be `convoCopy`:
```suggestion
      const convoCopy = { ...convo };
      convoCopy.messages.push(message);
      convoCopy.latestMessageText = message.text;

      return convoCopy;
```